lib/semantics/expression.cc

```diff
@@ -1822,9 +1822,8 @@ MaybeExpr ExpressionAnalyzer::Analyze(
 }
 
 void ExpressionAnalyzer::Analyze(const parser::CallStmt &callStmt) {
-  auto expr{AnalyzeCall(callStmt.v, true)};
-  if (auto *procRef{UnwrapExpr<ProcedureRef>(expr)}) {
-    callStmt.typedCall.reset(new ProcedureRef{*procRef});
+  if (auto expr{AnalyzeCall(callStmt.v, true)}) {
```

Comment:
Are you sure that this part of the change is needed?  The test on `expr` should not be required; the `Unwrap` routines should handle vacant optional values.  And the result of the `Unwrap` should not be dereferenced unconditionally -- please use a `DEREF` at least.

Reply:
Right, it's better to check the result of UnwrapExpr than of AnalyzeCall. I've fixed it.